### PR TITLE
Modify tests for TPCH explain plans to avoid regressions

### LINKED ISSUE #1377: Add tests for TPCH explain plans to avoid regressions
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
We have had performance regressions in the TPCH plans due to plan changes (e.g. https://github.com/apache/arrow-datafusion/issues/1367)

In order to avoid this situation in the future, we would like to have the explain plans committed as tests (so we can evaluate any changes to those plans as part of normal PR review)

**Describe the solution you'd like**
Add tests that `EXPLAIN ...` the TPCQ queries,  following the model of the test added by @viirya in  https://github.com/apache/arrow-datafusion/pull/1368 

Ideally this would be in the `tpch` crate, as suggested in @Dandandan  https://github.com/apache/arrow-datafusion/pull/1368/files#r757892477

**Describe alternatives you've considered**

### PR DESCRIPTION
**Pull request by jiangyinzuo**
# Which issue does this PR close?

Closes #1377.

# Rationale for this change

I modify `async fn expected_plan()` to utilize `EXPLAIN ...` SQL statement to generate both logical and physical plans. There are only logical plans in `benchmarks/expected-plans/*.txt` before this commit.

# What changes are included in this PR?

test output files in `benchmarks/expected-plans/*.txt` and `async fn expected_plan()` in `tpch.rs`
# Are these changes tested?

yes

# Are there any user-facing changes?

no

### CODE REVIEW

**Comment by alamb** [comment]
Thank you @jiangyinzuo 

@andygrove  can you give this a look?

File: benchmarks/src/bin/tpch.rs [hunk 1]
```diff
@@ -525,13 +525,19 @@ mod tests {
         let mut actual = String::new();
         let sql = get_query_sql(query)?;
         for sql in &sql {
-            let df = ctx.sql(sql.as_str()).await?;
-            let plan = df.into_optimized_plan()?;
-            if !actual.is_empty() {
-                actual += "\n";
+            if sql.starts_with("select") {
```
Comment (jiangyinzuo) [inline benchmarks/src/bin/tpch.rs:528]: handle special query 15 which contains `create view` sql statement

File: benchmarks/src/bin/tpch.rs [hunk 2]
```diff
@@ -562,7 +568,10 @@ mod tests {
     }
 
     fn create_context() -> Result<SessionContext> {
```
Comment (jiangyinzuo) [inline benchmarks/src/bin/tpch.rs:570]: I fix the parameter `target_partitions` so that the generated plans are the same in different machines.

**Review by andygrove — APPROVED** [review]
LGTM. Thanks @jiangyinzuo

**Review by Dandandan — APPROVED** [review]

**Comment by Dandandan** [comment]
Thank you @jiangyinzuo

**Pull request merged**